File: 334. Increasing Triplet Subsequence.py
import sys
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        x1, x2 = sys.maxint, sys.maxint
        for i in nums:
            if i < x1:
                x1 = i
            elif i > x1 and i < x2:
                x2 = i
            elif i > x2:
                return True
        return False

File: 334. Increasing Triplet Subsequence.cpp
#include <climits>
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int x1 = INT_MAX, x2 = INT_MAX;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] < x1)
                x1 = nums[i];
            else if (nums[i] > x1 && nums[i] < x2)
                x2 = nums[i];
            else if (nums[i] > x2)
                return true;
        }
        return false;
    }
};
